### PR TITLE
swarm/api: improve FUSE build constraints, logging and APIs

### PR DESCRIPTION
**Pull request by fjl**
This PR:

* fixes the build on BSD and other 'exotic' platforms.
* cleans up excessive log messages in FUSE code.
* ensures that swarmfs methods in the console cannot be invoked with the 
   wrong number of arguments.

### CODE REVIEW

**Comment by fjl** [comment]
@jmozah please review this if you have time

**Comment by jmozah** [comment]
@fjl  Apologies for adding more work to you.
Thanks, I learnt a lot from your corrections.

**Comment by jmozah** [comment]
@fjl WHen are you planning to merge this... i have the rw support coming up and i want to base that on this..

**Comment by karalabe** [comment]
He asked you to review it :) To see if everything's ok :D

**Comment by karalabe** [comment]
He's waiting on your approval to merge it :D

**Comment by fjl** [comment]
@jmozah doesn't have write access to the repo and can't approve it. I'm waiting for @zelig here.

File: swarm/api/swarmfs_unix.go [hunk 1]
```diff
@@ -14,77 +14,86 @@
...
+	}
+	return err == fuse.ErrOSXFUSENotFound
+}
+
+// MountInfo contains information about every active mount
 type MountInfo struct {
-	mountPoint     string
-	manifestHash   string
+	MountPoint     string
+	ManifestHash   string
```
Comment (jmozah) [inline swarm/api/swarmfs_unix.go]: why capitalise these two?

Comment (fjl) [inline swarm/api/swarmfs_unix.go]: In Go, the case of the first letter determines whether an identifier is public or not. These fields need to be public so they can be seen by the JSON encoder.

Comment (jmozah) [inline swarm/api/swarmfs_unix.go]: aah.. used in JSON encoder... thanks..

**Pull request merged**